platforms/java/com/area9innovation/flow/FlowPassword.java

```diff
@@ -102,4 +106,68 @@ public static Boolean validateHash(String password, String hash) {
 			
 		return Arrays.equals(correctpbkdf2, requestedpbkdf2);
 	}
+	
+	public static String getPasswordValidationString(int exp) {
+		Integer expLocal = exp < 0 ? null : ( exp == 0 ? VALIDATION_DEFAULT_EXP : exp);
+
+		byte[] bytes = new byte[PBKDF2_RANDOM_PASSWORD_BYTE_SIZE];
+		new SecureRandom().nextBytes(bytes);
+
+		String validation = "";
+
+		try {
+			JSONObject jo = new JSONObject();
+			jo.put("key", Base64.getEncoder().encodeToString(bytes));
+			if (expLocal != null) {
+				long unixTime = System.currentTimeMillis() / 1000L;
+				jo.put("exp", unixTime + expLocal);
+			};
+
+			validation = strtr(Base64.getEncoder().encodeToString(jo.toString().getBytes()), "+/=", "*_-");
```

Comment:
I think you can just do "Base64.getUrlDecoder", that way you do not need the strtr function (If you need it, rename it to something like replaceCharaters

Reply:
I think we need this to produce the same string as security.php do. So string from java & php are the same and can be processed in both targets

Reply:
Right. Agree, the * and _ are not normal base64url encode.